### ISSUE TITLE
Error messaging needs to be improved

### ISSUE THREAD

**Issue by kriston13** [googleapis/nodejs-vision#515]
I'm using this library at the moment, and using the `client.asyncBatchAnnotateImages()` functionality, I get the following error message:
```
{"code":13,"details":"","metadata":{"internalRepr":{},"options":{}},"note":"Exception occurred in retry method that was not classified as transient"}
```

Looking at the full stack trace, I see that this is the gRPC module chucking an error:
```
{ Error: 13 INTERNAL: 
    at Object.callErrorFromStatus (/Users/jonathantan/Documents/SRC_code/AS/ml-labelling/node_modules/@grpc/grpc-js/build/src/call.js:30:26)
    at Http2CallStream.call.on (/Users/jonathantan/Documents/SRC_code/AS/ml-labelling/node_modules/@grpc/grpc-js/build/src/client.js:96:33)
    at Http2CallStream.emit (events.js:187:15)
    at process.nextTick (/Users/jonathantan/Documents/SRC_code/AS/ml-labelling/node_modules/@grpc/grpc-js/build/src/call-stream.js:97:22)
    at process.internalTickCallback (internal/process/next_tick.js:70:11)
  code: 13,
  details: '',
  metadata: Metadata { internalRepr: Map {}, options: {} },
  note:
   'Exception occurred in retry method that was not classified as transient' }
```

Problem is, this doesn't help at all with debugging the problem. I believe the error messages need to be relevant to the actual vision functionality. 

PS: I'm able to use the `image.labelDetection()` method successfully, so it's not an account/credentials thing.

**Comment by alexander-fenster** [comment]
Hi @kriston13,

This error is coming from the transport layer and may or may not be related to the vision API. 
 Normally, if it's a usage problem (invalid parameters and stuff like that), the error messages are quite relevant. `13 INTERNAL` means something just it not right at all (one example where I saw `13 INTERNAL` was when the TCP connection was reset in a very weird way during the data transfer).

Before we debug further, can you try switching to the legacy gRPC library and check if it works with it? To do that, install `grpc`:

```sh
$ npm install grpc
```

and pass it to the client constructor:
```js
const vision = require('@google-cloud/vision');
const grpc = require('grpc');
const client = new vision.ImageAnnotatorClient({ grpc });
```

Also, where is the code being executed? Is it your local machine, or a cloud function, or something else?

**Comment by kriston13** [comment]
Hi Alexander,

Thank you for your response.
We did manage to figure it out through trial & error, and we THINK it was a
permissions problem.

The code was being executed from our local development laptops, telling
cloud vision to look in a storage bucket for images to annotate, and to
write the output to the same bucket.
We think it was missing the storage write permissions because adding that
fixed it. Not 100% because - couldn't find any understandable error - but
we've progressed. :)

Thank you again
Cheers
